doc/config-generic-update-rollback/SONiC_Generic_Config_Update_and_Rollback_Design.md

```diff
@@ -374,85 +419,42 @@ There are a few SONiC applications which store its configuration in the ConfigDB
 }
 ```
 
-This table is used to restart corresponding systemd service when a patch operation has been perform on its corresponding ConfigDB table. 
-
-#### Stage-4 Add verification that services have absorbed the changes
-There is 2 types of services:
-- Manually absorbing changes through explicit service restart
-- Automatically abosorbing changes
-
-**Manually absorbing changes through explicit service restart**
-
-These are the services identified in [Stage-3 Identifying Services to Restart](#stage-3-identifying-services-to-restart). Just restarting the services and waiting for a success return should be enough
-
-**Automatically abosorbing changes**
+Although some other services do not need manual restarting, and absorb the configs silently, there is no good way to make sure there is no errors encountered while the services absorbed the changes. Some of these services might report an error to syslog, others might crash ...etc.
 
-There are some options:
-1) Do not verify at all, but how to make sure tables are updated in order which is required for [Stage-2 Patch Ordering](#stage-2-patch-ordering)
-2) Introduce a delay post any table update to make sure listening services have absorbed the change, but how to decide a delay for each service.
-3) Verify expected listening services to have abosorbed the change, maybe check if they were resarted after the table is updated?
-4) ... other options?
+Currently there is no standard approach of updating ConfigDB that takes care of manual service restart and verifying services have absorbed change correctly.
 
-#### Stage-5 Apply patch, service restarts and delays
-At this stage we have built all the steps needed. The steps include JsonPatch operations, restart operations and delay operations.
+So we are going to introduce a new contract overlaying ConfigDB changes, that will take care of making correct changes to ConfigDB as well as verifying that corresponding services have absorbed the changes correctly.
 
-The JsonPatch consistes of a list operation, and the operation follows this format:
+The contract would be:
```

Comment:
By the way, just out of curiosity, this "contract language" is taken out of some formal way of writing documents

Reply:
I made it up for important parts of a function: input, output, exceptions, side-effects, assumptions. As for the list of errors these are the similar to [http patch error](https://tools.ietf.org/html/rfc5789#section-2.2)